lib/src/otagrum/TabuList.hxx

```diff
@@ -0,0 +1,90 @@
+#ifndef OTAGRUM_TABULIST_HXX
+#define OTAGRUM_TABULIST_HXX
+
+#include <agrum/tools/graphs/DAG.h>
+#include <agrum/tools/core/list.h>
+#include <agrum/BN/learning/structureUtils/graphChange.h>
+
+#include <openturns/Sample.hxx>
+
+#include "otagrum/NamedDAG.hxx"
+#include "otagrum/CorrectedMutualInformation.hxx"
+#include "otagrum/IndicesManip.hxx"
+
+namespace OTAGRUM
+{
+
+class OTAGRUM_API TabuList : public OT::Object
+{
+public:
+  // Constructor
+  explicit TabuList(const OT::Sample &data, const NamedDAG &initial_dag,
+                    const unsigned int max_parents = 4, const unsigned int restarts = 1,
+                    const unsigned int tabu_list_size = 2);
+  explicit TabuList(const OT::Sample &data,
+                    const unsigned int max_parents = 4,
+                    const unsigned int restarts = 1,
+                    const unsigned int tabu_list_size = 2);
```

Comment:
you could use OT::UnsignedInteger instead of int (always 32 bits)

Reply:
It's done.
Do you want me to do the same for the remainder of the file ?
Like for the attributes for example.

Reply:
please